WalletWasabi/WabiSabi/Backend/WabiSabiConfig.cs

```diff
@@ -164,6 +166,14 @@ public WabiSabiConfig(string filePath) : base(filePath)
 	[JsonProperty(PropertyName = "AllowP2trOutputs", DefaultValueHandling = DefaultValueHandling.Populate)]
 	public bool AllowP2trOutputs { get; set; } = false;
 
+	[DefaultValue(Constants.FallbackCoordinatorSignerKeyHex)]
+	[JsonProperty(PropertyName = "CoordinatorSignerKeyHex", DefaultValueHandling = DefaultValueHandling.Populate)]
+	public string CoordinatorSignerKeyHex { get; set; } = Constants.FallbackCoordinatorSignerKeyHex;
```

Comment:
What do you suggest?

Reply:
The problem with Signer is that it is too generic and even with some context it is not immediately clear what it is for. 

In this case, while the Affiliation Server is authenticated with a certificate, the coordinator uses this as a poor's man authentication mechanism. I say poor's man because the most common way to do this kind of things is by pushing the authentication to the infrastructure level and use a certificate on the coordinator's side too.

I think it worth to come back to this later. Imo we should have to have strong reasons to implement things like this at application level.

In the meanwhile a better name could be something like AffiliationMessageSigner or similar.

Reply:
> In this case, while the Affiliation Server is authenticated with a certificate, the coordinator uses this as a poor's man authentication mechanism. I say poor's man because the most common way to do this kind of things is by pushing the authentication to the infrastructure level and use a certificate on the coordinator's side too.

> I think it worth to come back to this later. Imo we should have to have strong reasons to implement things like this at application level.

I don't agree with you:
  * This has nothing to do with quality of the code. It is a matter of the protocol between the coordinator and the affiliation server. Moving the authentication to the transport layer changes the protocol.
  * The reason why the protocol is designed in this way is we want the affiliate server to have a short proof of what the coordinator requested without having to remember the whole communication in transport layer.
  * This approach is not definitely not unusual. [Ownership proofs](https://github.com/satoshilabs/slips/blob/master/slip-0019.md), for example, use a very similar concept.

> In the meanwhile a better name could be something like AffiliationMessageSigner or similar.

I don't think so. To be precise, `CoordinatorSignerKeyHex` is a private (signing) key of a coordinator that is first DER encoded (to bytes) and then hex encoded (to string).

Reply:
> The reason why the protocol is designed in this way is we want the affiliate server to have a short proof of what the coordinator requested without having to remember the whole communication in transport layer.

If you move authentication to the transport layer you know that whatever reaches your server is authentic. You can then log the requests (not the traffic).

Otherwise, why do you think it is important for you to have a proof of what the coordinator requested but it is not important for the coordinator to have a proof of what the affiliation server to responded? Should the affiliation client know the affiliation server public key to be able to verify the signature of the coinjoinRequest response?

> I don't think so. To be precise, CoordinatorSignerKeyHex is a private (signing) key of a coordinator that is first DER encoded (to bytes) and then hex encoded (to string).


Yes, sorry, i was talking about the Signer class and not the private key here. Once you find a better name for the Signer the better name for this will become easier to find.

Reply:
Please change the name. I will not block this for a silly renaming but it would be good to have a name that help us to understand what we are reading. Make some reference to the purpose of the key at least, here my ugly try: AffiliationMessageSigner for the signer and AffiliationMessageSignerKey for the key.

Reply:
Done in https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/80baeacf449ed871d3bdd8f89950a5805c3dc0b3, https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/4edcf80dedba16cfb9f3599f98ed7b5d4fa919f4 (and https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/9b512968e7632b5f6f2ef078049ad7e34b508269, https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/dc39f3debad8e95c6f40f30498fbb8235a59a4c8, https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/38ea11994292722d4fab2e38ec9ac2c450f35987, https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/c3ba0079078564790dc0737e014b283395457a26, https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/ab9b5dc093b8fe9325ba0b58fcfb119cc4d62a80).